lib/discover/local.go

```diff
@@ -281,3 +288,30 @@ func (c *localClient) registerDevice(src net.Addr, device Announce) bool {
 
 	return isNewDevice
 }
+
+// filterUnspecifiedLocal returns the list of addresses after removing any
+// unspecified, localhost, multicast, broadcast or port-zero addresses.
+func filterUnspecifiedLocal(addrs []string) []string {
+	filtered := addrs[:0]
+	for _, addr := range addrs {
+		u, err := url.Parse(addr)
+		if err != nil {
+			continue
+		}
+
+		tcpAddr, err := net.ResolveTCPAddr("tcp", u.Host)
+		if err != nil {
+			continue
+		}
+
+		switch {
+		case len(tcpAddr.IP) == 0:
```

Comment:
Do we not "fill" the unspecified address on the other side? We could continue to send them as long as we fill them?

Reply:
We don't, and I don't think we can as the local discovery packet doesn't come from a bep listening socket.

Reply:
I guess in principle we could send it from the quic socket, if we listen on quic, but we currently don't. (All this predates quic support by a wide margin.)